### ISSUE TITLE
Support Plugins

### ISSUE THREAD

**Issue by yassinebridi** [redwoodjs/graphql#388]
I know redwood is trying to make every major operation opinionated, but supporting plugins would open a whole new world of possibilities, there definitely would be areas that redwood might not think about implementing to the core functionality, for example:

- An image processor for optimizations.
- Different ways of deploying, giving the ability maybe to not well-know hosting providers to implement their own solutions for redwood.

There could be an official store for the plugins that has been reviewed thoroughly, and has respected the high standards that redwood would outline.

**Comment by guledali** [comment]
I saw today that someone have already made a plugin for `redwoodjs`, https://github.com/freddydumont/redwood-nprogress

There has to be somewhere where you can find all plugins like centralized repo/store similar to gatsbyjs https://www.gatsbyjs.org/plugins/

**Comment by yassinebridi** [comment]
Exactly, and adding support to the redwood cli, from listing to installing/uninstalling plugins etc..

**Comment by guledali** [comment]
It would be nice if redwoodjs forced library authors to think of how people will write tests with their plugins. For example both devise and CarrierWave have good docs on how write tests with them.

Redwoodjs should enforce testing culture or at very least encourage it.

**Comment by jtoar** [comment]
Just wanted to add that this is something we're actively thinking about now.
